netbox_netprod_importer/vendors/cisco/ios.py

```diff
@@ -108,6 +108,38 @@ def get_detailed_cdp_neighbours(self):
         for n in neighbours:
             yield n
 
+    def get_detailed_lldp_neighbours(self):
+        cmd = "show lldp neighbors detail"
+        cmd_output = self.device.cli([cmd])[cmd]
+        cmd_output = re.split('---+\n', cmd_output)
+        neighbours = []
+        for lldp_port_info in cmd_output:
+            if len(lldp_port_info):
+                info_port = re.split('\n\n', lldp_port_info)[0].split('\n')
+                hostname = local_port = port = ''
+                for line in info_port:
+                    value = line.split(':')[1].strip()
+                    if line.startswith('System Name'):
+                        hostname = value
+                    elif line.startswith('Local Intf'):
+                        local_port = value
+                    elif line.startswith('Port id'):
+                        port = value
+                    elif line.startswith('Chassis id'):
+                        chassis_id = value
+
+                neighbours.append(
+                    {
+                        "local_port": local_port,
+                        "hostname": hostname,
+                        "port": port,
+                        "chassis_id": chassis_id
+                    }
+                )
```

Comment:
I would prefer having the neighbours attributes (local_port, hostname, port, chassis_id) into a dictionary that you then append to the neighbours list, instead of different variables that are declared in the scope of the for (it's apparently working in python, but it's unintuitive).

Reply:
Also, instead of appending all the neighbours in a list and yielding at the end, you can yield each neighbour at line 131.